### PR TITLE
Silence pedantic compiler warnings.

### PR DESCRIPTION
**Pull request by samuelkgutierrez**
Signed-off-by: Samuel K. Gutierrez <samuel@lanl.gov>

### CODE REVIEW

File: src/common/pmix_query.c [hunk 1]
```diff
@@ -984,7 +985,7 @@ static pmix_query_t * pmix_query_strip_local_keys(pmix_query_t orig_queries[],
                                                   size_t orig_nqueries,
                                                   size_t nqueries)
 {
-    int rc;
+    int rc = PMIX_SUCCESS;
```
[the remark this answers is not in the document]
Comment (samuelkgutierrez) [reply src/common/pmix_query.c:988]: Hi Ralph,

It's hidden here: https://github.com/samuelkgutierrez/openpmix/blob/silence-compiler-warnings/src/common/pmix_query.c#L998

**Comment by samuelkgutierrez** [comment]
bot:lanl:retest

File: src/common/pmix_query.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (rhc54) [reply src/common/pmix_query.c:988]: Then I'm confused - why are you adding the `PMIX_HIDE_UNUSED_PARAMS`??

[the remark this answers is not in the document]
Comment (rhc54) [reply src/common/pmix_query.c:988]: I see - we really shouldn't be using "HIDE" for that purpose. If the compiler is complaining about the return code not being used/checked, then let's check it. If it isn't success, then free the queries and return NULL.

**Pull request merged**